### PR TITLE
Roger/interface sketch

### PR DESCRIPTION
**Pull request by RogerQi**
* contains changes from alvin/download

* revoke low res TSDF setting

* suppress imgui examples that are causing troubles

* add draft top level interface

* update interfacing toplevel module

* add data gatherer

* add placeholder README

### CODE REVIEW

File: interface/top_level.h [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+ * 
+ * @var H           an unsigned integer indicating the height of the image.
+ * 
+ * @var W           an unsigned integer indicating the width of the image.
+ * 
+ * @var timestamp   a floating number containing UNIX timestamp of the image
+ *                  indicating when it was captured by the perception module.
+ */
+typedef struct {
+    cv::Mat* data;
```
Comment (alvinsunyixiao) [inline interface/top_level.h:23]: This is not safe if we pass around threads. Why make this a pointer?

File: interface/top_level.h [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
```
Comment (alvinsunyixiao) [inline interface/top_level.h:41]: Same as above

File: interface/top_level.h [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+ *                  R =  [R[3], R[4], R[5]]     t = (t[0], t[1], t[2])^{T}
+ *                       [R[6], R[7], R[8]]]
+ * 
+ * @var timestamp   a floating number containing a UNIX timestamp indicating the capture
+ *                  time of the corresponding ZED images that were used to estimate
+ *                  this pose.
+ */
+typedef struct {
+    float R[9];
+    float t[3];
```
Comment (alvinsunyixiao) [inline interface/top_level.h:63]: Why not reuse the `SE3` class in `lie_groups.cuh`?

File: interface/top_level.h [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    float fx;
+    float fy;
+    float cx;
+    float cy;
+    float k1;
+    float k2;
+    float k3;
+    float p1;
+    float p2;
+    float depth_scale;
```
Comment (alvinsunyixiao) [inline interface/top_level.h:107]: Why is this depth scale part of a camera intrinsics?

**Comment by RogerQi** [comment]
Top level stuffs are removed in later commits.

**Review by alvinsunyixiao — APPROVED** [review]
LGTM -- though this codebase is becoming messier and messier :(

**Comment by RogerQi** [comment]
Do you have any suggestion for keeping it less messier? Also, I think doing rebase and merge instead of squash and merge may be a better idea. What do you think

**Comment by alvinsunyixiao** [comment]
Rebase merge is a little more error-prone. In terms of code quality, there're many many stuff needed to be clean up. Let's find some separate time for that.

Suggestions for merging:
I think its better to merge the frontier first. For example if we have three feature branches A, B, and C
master -> A -> B -> C

Then we should setup three PRs and review separately
1. C based B
2. B based A
3. A based master

then we should consider same order for merging, i.e.
1. C into B
2. B into A
3. A into master

**Pull request merged**